src/assets/assets.cpp

```diff
@@ -1621,6 +1621,50 @@ bool GetAssetFromCoin(const Coin& coin, std::string& strName, CAmount& nAmount)
     return false;
 }
 
+void GetAssetData(const CScript& script, CAssetOutputEntry& data)
+{
+    // Placeholder strings that will get set if you successfully get the transfer or asset from the script
+    std::string address = "";
+    std::string assetName = "";
+
+    // Get the New Asset or Transfer Asset from the scriptPubKey
+    if (script.IsNewAsset()) {
+        CNewAsset asset;
+        if (AssetFromScript(script, asset, address)) {
+            assetName = asset.strName;
+            data.type = "new_asset";
```

Comment:
These type strings (new_asset, reissue_asset, transfer_asset) are used in a few places in different files to coordinate and could be moved into constants.

Reply:
Done